File: lib/features/auth/ui/login_screen_data.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'login_screen_data.freezed.dart';

@freezed
sealed class LoginScreenData with _$LoginScreenData {
  const factory LoginScreenData.phoneEntry({
    required bool canSend,
    required bool sending,
  }) = LoginPhoneEntry;

  const factory LoginScreenData.otpEntry({
    required String phone,
    required bool canVerify,
    required bool verifying,
  }) = LoginOtpEntry;
}

File: app/src/main/java/com/billpayments/features/auth/ui/LoginScreenData.kt
package com.billpayments.features.auth.ui

import androidx.compose.runtime.Immutable

@Immutable
sealed interface LoginScreenData {
    data class PhoneEntry(val phone: String, val canSend: Boolean, val sending: Boolean) : LoginScreenData
    data class OtpEntry(val phone: String, val otp: String, val canVerify: Boolean, val verifying: Boolean) : LoginScreenData
}
